Core/CRFSharpWrapper/CRFSharpWrapper.csproj

```diff
@@ -66,6 +66,7 @@
   <ItemGroup>
     <Compile Include="Args.cs" />
     <Compile Include="CRFSharpHelper.cs" />
+    <Compile Include="DecoderDriver.cs" />
```

Comment:
Thanks. Could you please merge it into Decoder.cs rather than creating a separated file ? Then both encoder and decoder in wrapper will have same style.

Reply:
Yes, that was my initial thought but currently Decoder.cs has a lower-level API (LoadModel, CreateTagger, Segment) and I didn't want to mix these two different-level APIs (as dictated by the [single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle)).

The alternative would be to rename Decoder but that would break backward compatibility. To be honest, I highly doubt that anyone is using the Decoder API as it's horrible (sorry for a strong word but it is!). The decoding code from the README simply doesn't work. Personally I wouldn't mind breaking or losing that API.

Would you like me to rename Decoder to something else (LowLevelDecoder?) and move Decode(args) to Decoder?

Reply:
Yes. Probably both Encoder and Decoder need to be refactored. Maybe we should split them into Encoder, EncoderWrapper, Decoder and DecoderWrapper. Encoder and Decoder have detailed internal APIs, and EncoderWrapper and DecoderWrapper provide high level APIs to users.